packages/svelte-ux/src/lib/utils/number.ts

```diff
@@ -1,45 +1,109 @@
-import { format as d3Format } from 'd3-format';
+import { format as d3Format, formatDefaultLocale, type FormatLocaleDefinition } from 'd3-format';
+
+export type FormatNumberStyle =
+  | 'decimal' // from Intl.NumberFormat options.style NumberFormatOptions
+  | 'currency' // from Intl.NumberFormat options.style NumberFormatOptions
+  | 'percent' // from Intl.NumberFormat options.style NumberFormatOptions
+  | 'unit' // from Intl.NumberFormat options.style NumberFormatOptions
+  | 'none'
+  | 'integer'
+  | 'percentRound'
+  | 'metric';
+
+export type FormatNumberOptions = Intl.NumberFormatOptions & {
+  style?: FormatNumberStyle;
+  locales?: string | undefined;
+  fractionDigits?: number;
+};
 
 // See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat
 export function formatNumber(
   number: number | null | undefined,
-  options: Intl.NumberFormatOptions & { fractionDigits?: number } = {}
+  options: FormatNumberOptions & {
+    suffix?: string;
+    /**
+     * If number is >= 2, then this extraSuffix will be appended
+     * @default 's'
+     */
+    suffixExtraIfMany?: string;
+  } = {}
 ) {
   if (number == null) {
     return '';
   }
 
-  const formatter = Intl.NumberFormat(undefined, {
+  if (options.style === 'none') {
+    return `${number}`;
+  }
+
+  if (options.style === 'integer') {
+    return `${parseInt(number.toString())}`;
```

Comment:
This should also round so `1234.56` becomes `1,235`

Reply:
let's speak about all this on the previous comment, I revolve this one